### ISSUE TITLE
Model reproducible without set seeds though subsampling, colsample_bytree and by level are < 1

### ISSUE THREAD

**Issue by Zethson** [dmlc/xgboost#5628]
Dear developers,

I am currently just for the sake of evaluation training a useless model on the Boston dataset.
Here, I noticed that even if I do not set the XGBoost random seed, the model is somehow reproducible.

XGBoost 1.0.2

25 epochs

Code:

```
import click
import xgboost as xgb
import numpy as np
from sklearn.datasets import fetch_covtype, load_boston
from sklearn.model_selection import train_test_split
import time
import random
import os


@click.command()
@click.option('--seed', type=int, default=0)
@click.option('--epochs', type=int, default=25)
@click.option('--no-cuda', type=bool, default=False)
def train(seed, epochs, no_cuda, dataset='covertype):
    # Fetch dataset using sklearn
    dataset = fetch_covtype()
    param = {
        'objective': 'multi:softmax',
        'num_class': 8
        # 'single_precision_histogram': True
    }

    X = dataset.data
    y = dataset.target

    # Create 0.75/0.25 train/test split
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, train_size=0.75, random_state=0)

    param['subsample'] = 0.5
    param['colsample_bytree'] = 0.5
    param['colsample_bylevel'] = 0.5

    # Convert input data from numpy to XGBoost format
    dtrain = xgb.DMatrix(X_train, label=y_train)
    dtest = xgb.DMatrix(X_test, label=y_test)

    # Set CPU or GPU as training device
    if no_cuda :
        param['tree_method'] = 'hist'
    else:
        param['tree_method'] = 'gpu_hist'

    # Train on the chosen device
    results = {}
    gpu_runtime = time.time()
    xgb.train(param, dtrain, epochs, evals=[(dtest, 'test')], evals_result=results)
    if not no_cuda:
        print(f'GPU Run Time: {str(time.time() - gpu_runtime)} seconds')
    else:
        print(f'CPU Run Time: {str(time.time() - gpu_runtime)} seconds')

if __name__ == '__main__':
    train()
```

**Comment by Zethson** [comment]
Note: If I do set the seeds and change them, the model is affected.
Is there a default seed set or how does this happen?

**Comment by trivialfis** [comment]
Yes, there s a default seed.
